### PR TITLE
Visualize the inline breakpoint locations

### PR DESCRIPTION
**Pull request by testforstephen**
What this PR does:
- Support BreakpointLocations DAP Request, this can visualize the locations where inline breakpoints can be added. Closes microsoft/vscode-java-debug#1193.
image

- Show an extra column cursor when an inline breakpoint is hit. Closes microsoft/vscode-java-debug#1202
image

- Line breakpoint will only break on the first location on that line. If the line contains other lambdas, you can add an inline breakpoint to break on other locations.

### CODE REVIEW

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java [hunk 1]
```diff
@@ -149,6 +173,7 @@ public void setHitCount(int hitCount) {
             .filter(request -> request instanceof BreakpointRequest)
             .subscribe(request -> {
                 request.addCountFilter(hitCount);
+                request.disable();
```
Comment (jdneo) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java:176]: why we need to call `disable` first here?

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java [hunk 2]
```diff
@@ -181,24 +229,56 @@ public String[] getFullyQualifiedName(String uri, int[] lines, int[] columns) th
...
+                    sourceLocation.setMethodSignature(locator.getMethodSignature());
                 }
             }
         }
-        return fqns;
+
+        return sourceLocations;
+    }
+
+    private BreakpointLocation[] getInlineBreakpointLocations(final CompilationUnit astUnit, int sourceLine) {
```
Comment (jdneo) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java:260]: Besides the lambda expression. Do we support multiple statements in one line? for example: 

```java
int a = 0; int b = 1; int c = a+b;
```

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java (hunk 1 above)
Comment (testforstephen) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java:176]: image

Calling request.enable() will do nothing if the request has been enabled before. We need to invalidate the old request, and re-enable it to send the new JDWP request to debuggee.

https://github.com/eclipse-jdt/eclipse.jdt.debug/blob/96cda748fde8d874fff6e3fad1f3f775377eac93/org.eclipse.jdt.debug/jdi/org/eclipse/jdi/internal/request/EventRequestImpl.java#L245-L248

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java (hunk 2 above)
Comment (testforstephen) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java:260]: We only support adding an inline breakpoint for lambda only. This is also the behavior of IJ.

**Review by jdneo — APPROVED** [review]

**Pull request merged**